lib/common/memory/src/madvise.rs

```diff
@@ -169,3 +170,29 @@ fn populate_simple(slice: &[u8]) {
             .sum::<Wrapping<u8>>(),
     );
 }
+
+/// For given file path, clear disk cache with `posix_fadvise`
+///
+/// If `posix_fadvise` is not supported, this function does nothing.
+pub fn clear_disk_cache(file_path: &Path) -> io::Result<()> {
+    #[cfg(target_os = "linux")]
+    {
+        use std::os::unix::io::AsRawFd;
+
+        use libc::{POSIX_FADV_DONTNEED, posix_fadvise};
+
+        let file = std::fs::OpenOptions::new().read(true).open(file_path)?;
+        let fd = file.as_raw_fd();
+
+        // Clear the cache, offset 0, length 0 (whole file)
+        let ret = unsafe { posix_fadvise(fd, 0, 0, POSIX_FADV_DONTNEED) };
+
+        if ret != 0 {
+            Err(io::Error::from_raw_os_error(ret))
+        } else {
+            Ok(())
+        }
+    }
+    #[cfg(not(target_os = "linux"))]
+    Ok(())
+}
```

Comment:
We have a similar function here:
https://github.com/qdrant/qdrant/blob/4db98ecdbf66bedbce268ec26865ea446d0bba7b/lib/common/io/src/file_operations.rs#L34-L35

Reply:
`advise_dontneed` doesn't seem to be used anywhere. I propose to remove it and keep a new one.

Reply:
Then I propose to copy the old definition:
- It uses safe wrapper `nix::fcntl` instead of unsafe `libc` call.
- It works on any platform that support `posix_fadvise`, not just Linux.

Reply:
Done